### PR TITLE
[Python] Fail WindowInto when there are no inputs to it

### PR DESCRIPTION
**Pull request by riteshghorse**
It has been observed that pipelines get stuck when there are no input PCollections to a beam.WindowInto. No logs are observed in the console and the only option left is to force terminate jobs. This PR catches that case and hard fails the pipeline at submission time.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by riteshghorse** [comment]
Run Python 3.8 PostCommit

**Comment by riteshghorse** [comment]
R: @tvalentyn

**Comment by robertwb** [comment]
FWIW, I think this should be fixed in the service rather than adding a special case to the SDK (Python only, which won't be out for 6 weeks or so).

**Comment by robertwb** [comment]
A good place to validate this might be https://github.com/apache/beam/blob/release-2.51.0/sdks/python/apache_beam/runners/common.py#L1895

We could have a mapping of known URNs -> expected input/output cardinality (if any), and flag violations. (We could ensure the inputs/outputs actually exist.)

`p | beam.WindowInto(...)` should also be an error right there (checked in WindowInto's expand method).

File: sdks/python/apache_beam/pipeline.py [hunk 1]
```diff
@@ -1353,18 +1353,28 @@ def transform_to_runner_api(
       environment_id = context.get_environment_id_for_resource_hints(
           self.resource_hints)
 
+    inputs = {
```
Comment (tvalentyn) [inline sdks/python/apache_beam/pipeline.py:1356]: > p | beam.WindowInto(...) should also be an error right there (checked in WindowInto's expand method).

+1 to Robert's comment, moving this validation inside `WindowInto` methods  would be better.

File: sdks/python/apache_beam/runners/common.py [hunk 2]
```diff
@@ -1928,6 +1928,10 @@ def validate_transform(transform_id):
             "Incompatible input coder %s and output coder %s for transform %s" %
             (transform_id, input_coder, output_coder))
 
+    if transform_proto.spec.urn == common_urns.primitives.ASSIGN_WINDOWS.urn:
+      if not transform_proto.inputs:
+        raise ValueError("Unexpected number of inputs: %s" % transform_proto)
```
Comment (robertwb) [inline sdks/python/apache_beam/runners/common.py:1933]: Maybe say "Missing inputs?"

File: sdks/python/apache_beam/runners/common.py [hunk 3]
```diff
@@ -1928,6 +1928,10 @@ def validate_transform(transform_id):
             "Incompatible input coder %s and output coder %s for transform %s" %
             (transform_id, input_coder, output_coder))
 
+    if transform_proto.spec.urn == common_urns.primitives.ASSIGN_WINDOWS.urn:
```
Comment (robertwb) [inline sdks/python/apache_beam/runners/common.py:1931]: Let's validate `common_urns.primitives.PAR_DO.urn` as well.

**Pull request merged**